modules/ROOT/pages/datagraph-cli-errors.adoc

```diff
@@ -0,0 +1,227 @@
+* `datagraph sources promote`: The source ID or the target environment ID are not in a valid UUID format.
+
+=== API Already Exists
+
+Output:
+----
+{
+  "errorCode": "dg-federation-source-exists",
+  "errorMessage": "API already exists"
+}
+----
+
+Cause:
+
+This error indicates that the API you are trying to add already exists in the unified schema.
+
+* `datagraph sources add`: The asset ID and major version of the API source already exists in the unified schema. To make changes to the source, use the the `datagraph sources update` command.
+
+=== No API Instances Found
+
+Output:
+----
+{
+  "errorCode": "dg-xapi-no-api-instances",
+  "errorMessage": "No API instances found"
+}
+----
+
+Cause:
+
+This error indicates the lack of URL to retrieve for an API instance.
+
+* `datagraph sources add/update/promote`: This errors appears when using the ``--discover-url` option. It occurs when the asset has no valid instances from which the URL can be retrieved that aren’t from a mocking service. In such cases, use the `--url` option instead.
+
+=== Must Define URL
+
+Output:
+----
+{
+  "error": {
+    "errorMessage": "Must define url or set --discover-url"
+  }
+}
+----
+
+Cause:
+
+This error indicates that you must set the URL using either `--url` or `--discover-url`.
+
+* `datagraph sources add`: Since an URL is a required field in the `add` command, you must set one of the two options, `--url` or `--discover-url`.
+
+=== Invalid API Calls
+
+Output:
+----
+{
+  "errorCode": "dg-xapi-invalid-api-calls",
+  "errorMessage": "Invalid Api Calls: 1500"
+}
+----
+
+Cause:
+
+This error indicates that you requested an invalid number of API calls.
+
+* `datagraph scale`: The number of concurrent calls must be a number between 1 and 1200.
+
+=== Unauthorized
+
+Output:
+----
+{
+  "error": {
+    "code": "401",
+    "message": "Unauthorized"
+  }
+}
+----
+
+Cause: 
+
+This error indicates that you lost the session in the CLI when using it in interactive mode. Exit and relaunch the CLI, and then log in again.
+
+=== API Not Found in Current Environment
+
+Output:
+----
+{
+  "errorCode": "dg-federation-source-not-in-env",
+  "errorMessage": "API not found in current environment"
+}
+----
+
+Cause:
+
+This error indicates that the API source you are trying to edit, delete, or update exists, but it is not in the working environment.
+
+* `datagraph sources update/promote`: The UUID sent is not in the working environment. Or if you are using the `--version` option, the major version is not compatible with the current version.
+
+== Validation Errors
+
+Validation errors occur when you use the `datagraph validate asset` command before adding or updating an API source against the unified schema.
+
+=== Editing Errors
+
+Editing errors are generated when the edits in the particular API source contain a semantic problem. These errors use the code property `dg-federation-customization-conflict` and have the following structure:
```

Comment:
Editing errors are generated when > DataGraph returns these errors when... (to sync with == Command Errors)